### PR TITLE
Update membership

### PR DESCRIPTION
**Pull request by eshark9312**
- #4330 

Add section to edit membership modal for validator's membership.
Add additional step to edit membership transaction modal for bond/unbond validator's account to the membership.

[> Design](https://www.figma.com/file/QnjZuWYtGimTEkQlN6MNlH/Validators?type=design&node-id=1717-236794&mode=design&t=z1LVePeYx80EU2R5-0)

### CODE REVIEW

File: packages/ui/src/memberships/components/MemberListItem/MyMember.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ export const MyMemberListItem = ({ member }: { member: Member }) => {
         <CountInfo count={invited} />
       </MemberColumn>
       <MemberControls>
-        <EditMembershipButton member={member} size="small">
+        <EditMembershipButton member={member} size="small" className="edit-button">
```
Comment (thesan) [inline packages/ui/src/memberships/components/MemberListItem/MyMember.tsx:50]: Is this class used anywhere ?

File: packages/ui/src/app/pages/Validators/ValidatorList.stories.tsx [hunk 2]
```diff
@@ -456,7 +456,7 @@ export const TestsFilters: Story = {
       await selectFromDropdown(screen, verificationFilter, 'verified')
       await waitFor(() => expect(screen.queryAllByRole('button', { name: 'Nominate' })).toHaveLength(3))
       expect(screen.queryByText('unverifed')).toBeNull()
-      expect(screen.getByText('alice'))
+      expect(screen.getAllByText('alice').length).toEqual(2)
```
Comment (thesan) [inline packages/ui/src/app/pages/Validators/ValidatorList.stories.tsx:459]: Why is Alice appearing twice in the list now ?

File: packages/ui/src/app/pages/Proposals/CurrentProposals.stories.tsx [hunk 3]
```diff
@@ -785,7 +785,7 @@ export const SpecificParametersFundingRequest: Story = {
 
 export const SpecificParametersMultipleFundingRequest: Story = {
   play: specificParametersTest('Funding Request', async ({ args, createProposal, modal, step }) => {
-    const bob = member('bob')
+    const dave = member('dave')
```
Comment (thesan) [inline packages/ui/src/app/pages/Proposals/CurrentProposals.stories.tsx:788]: Are these changes necessary ? It will conflict with #4713 when `dev` will be merged into `validators` (which I should have done earlier already but I forgot :facepalm:).

**Review by thesan — COMMENTED** [review]
@eshark9312 great work again I just have a few minor comments. If merging `dev` into `validators` will simplify things I can do it ASAP.

About the failing storybook tests my guess is that it's initially due to a Vercel shortage on Friday, but what's weird is that it still not picking up now so I'll try to debug this tomorrow.

**Comment by thesan** [comment]
@eshark9312 actually I went ahead with the merge already because to was blocking #4733. So now there's a bunch of conflicts on this PR. Sorry I should have done that earlier...

**Comment by thesan** [comment]
@eshark9312 I just confirmed that the storybook tests failure is a Vercel issue for some reason it didn't register linking the deployment to: https://pioneer-2-storybook-git-fork-eshark9312-update-bc4165-joystream.vercel.app (even though this domain works fine)
image

Re-deployments didn't fix this, maybe pushing new commits will. So please take care of the conflicts. If it persist I'll contact Vercel.

File: packages/ui/src/memberships/components/MemberListItem/MyMember.tsx (hunk 1 above)
Comment (eshark9312) [reply packages/ui/src/memberships/components/MemberListItem/MyMember.tsx:50]: Yes, it is used for storybook test:
https://github.com/Joystream/pioneer/blob/53677ace201ab0324346ac6bdba35ce303f04ade/packages/ui/src/app/pages/Profile/MyMemberships.stories.tsx#L258C4-L258C73

File: packages/ui/src/app/pages/Proposals/CurrentProposals.stories.tsx (hunk 3 above)
Comment (eshark9312) [reply packages/ui/src/app/pages/Proposals/CurrentProposals.stories.tsx]: I removed this change.

**Review by thesan — APPROVED** [review]
LGTM!
